frontend/src/components/Graph.tsx

```diff
@@ -82,34 +89,126 @@ const CustomGraph: React.FC = () => {
     }
   }
 
+  const timeBar = () => {
+    const graph = graphRef.current
+    const timeBarData: TimeBarData[] = []
+    if (graph) {
+      if ((graph!.save() as GraphData)!.nodes!.length > 0) {
+        ;(graph!.save() as GraphData)!.nodes!.forEach((node) => {
+          timeBarData.push({
+            date: String(node.time),
+            value: String(node.id),
```

Comment:
instead of doing graph.save, can you create a function where the node information is added to the timeBarData. This function can be called in the onMessage function where we iterate over all nodes in the graph anyways. It will reduce the complexity by a lot. If confused, look at how the layout function works.

Reply:
Fixed by latest commit